thoth-api/src/graphql/model.rs

```diff
@@ -1481,7 +1486,17 @@ impl MutationRoot {
 
         data.validate()?;
 
-        Work::create(&context.db, &data).map_err(|e| e.into())
+        let result = Work::create(&context.db, &data).map_err(|e| e.into());
+
+        if let Ok(created_work) = result.clone() {
```

Comment:
no need to `clone` here, you can just reference `created_work`: `if let Ok(ref created_work) = result`